### PR TITLE
job timeouts

### PR DESCRIPTION
**Pull request by jeyj0**
Adds an optional timeoutInMicroseconds.

Also automatically sets jobs that are set to "running" on startup to "failed" with a custom message, as that's more appropriate than setting them to time out as we talked about before.

### CODE REVIEW

File: IHP/Job/Runner.hs [hunk 1]
```diff
@@ -106,6 +108,9 @@ jobWorkerFetchAndRunLoop JobWorkerArgs { .. } = do
     let ?context = frameworkConfig
     let ?modelContext = modelContext
 
+    runningJobs <- Queue.fetchRunningJobs @job
+    mapM_ (Queue.jobDidFail' job "Job was running at startup") runningJobs
```
Comment (mpscholten) [inline IHP/Job/Runner.hs:112]: What happens if two job workers are started?

Comment (jeyj0) [reply IHP/Job/Runner.hs:112]: Oops, yeah, that shouldn't go there. I had something else there first, which worked with timeouts, where it actually made sense - but now it definitely shouldn't go here, I think. Will take a closer look

**Comment by jeyj0** [comment]
Ready for review again, @mpscholten

**Comment by mpscholten** [comment]
I think this still has the same problem: When a second worker process is started, the existing running jobs are killed

**Comment by jeyj0** [comment]
Wait - but isn't the code only run at startup, and before the workers are actually started?

**Comment by mpscholten** [comment]
Yes but when I start two different processes (e.g. two servers connected to one DB) it will kill the jobs of the other process

**Comment by jeyj0** [comment]
Ooooohh! Yes, that's a case I hadn't thought of. I'll take a look again...

**Comment by jeyj0** [comment]
Separated into two:
1. job timeouts, which has a new PR
2. cancel at startup, which requires some more thought

**Pull request closed without merging**